src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs

```diff
@@ -233,9 +247,24 @@ private CommandLineSourceRepositoryProvider GetSourceRepositoryProvider()
             var primaryRepositories = packageSources.Select(sourceRepositoryProvider.CreateRepository);
 
             var allowPrerelease = Prerelease || (version != null && version.IsPrerelease);
+            
+            DependencyBehavior depBehavior = GetSettingsDependencyBehavior(Settings);
```

Comment:
This does not need to be read if the command line flag is set, it should be moved to an else after that check below